src/testers.rs

```diff
@@ -1,30 +1,130 @@
+    }
+
+    if args_len > max {
+        return Err(TeraError::TestError(
+            tester_name.to_string(),
+            format!(
+                "{} was called on the variable {} with {} arguments, the max number is {}. ",
+                tester_name, arg_name, args_len, max
+            )
+        ))
+    }
+
+    Ok(())
+}
+
+// Called to return an error when unwrapping the value and realising there's nothing
+fn value_is_undefined(arg_name: &str, tester_name: &str) -> TeraResult<bool> {
+    Err(TeraError::TestError(
+        tester_name.to_string(),
+        format!("{} was called on the variable {}, which is undefined", tester_name, arg_name)
+    ))
+}
+
+/// Returns true if `value` is defined. Otherwise, returns false.
+pub fn defined(name: &str, value: Option<Value>, params: Vec<Value>) -> TeraResult<bool> {
+    try!(number_args_allowed(name, "defined", 0, params.len()));
+
+    Ok(value.is_some())
+}
+
+/// Returns true if `value` is undefined. Otherwise, returns false.
+pub fn undefined(name: &str, value: Option<Value>, params: Vec<Value>) -> TeraResult<bool> {
+    try!(number_args_allowed(name, "undefined", 0, params.len()));
+
+    Ok(value.is_none())
+}
+
+/// Returns true if `value` is a string. Otherwise, returns false.
+pub fn string(name: &str, value: Option<Value>, params: Vec<Value>) -> TeraResult<bool> {
+    try!(number_args_allowed(name, "string", 0, params.len()));
+
+    match value {
+        Some(Value::String(_)) => Ok(true),
+        None => value_is_undefined(name, "string"),
+        _ => Ok(false)
     }
+}
 
-    let name = match *value {
-        Node::Identifier { ref name, .. } => name,
-        _ => return Err(TeraError::TestError("defined".to_string(),
-                "defined can only be called on identifiers".to_string()))
-    };
+/// Returns true if `value` is a number. Otherwise, returns false.
+pub fn number(name: &str, value: Option<Value>, params: Vec<Value>) -> TeraResult<bool> {
+    try!(number_args_allowed(name, "number", 0, params.len()));
 
-    Ok(context.contains_key(name))
+    match value {
+        Some(Value::I64(_)) | Some(Value::F64(_)) | Some(Value::U64(_)) => Ok(true),
+        None => value_is_undefined(name, "number"),
+        _ => Ok(false)
+    }
 }
 
+/// Returns true if `value` is an odd number. Otherwise, returns false.
+pub fn odd(name: &str, value: Option<Value>, params: Vec<Value>) -> TeraResult<bool> {
+    try!(number_args_allowed(name, "odd", 0, params.len()));
+
+    match value {
+        Some(v) => match v.to_number() {
+            Ok(f) => Ok(f % 2.0 != 0.0),
+            Err(_) => Err(TeraError::TestError(
+                "odd".to_string(),
+                "odd can only be called on numbers".to_string()
+            ))
+        },
+        None => value_is_undefined(name, "odd")
+    }
+}
+
+
+/// Returns true if `value` is an even number. Otherwise, returns false.
+pub fn even(name: &str, value: Option<Value>, params: Vec<Value>) -> TeraResult<bool> {
```

Comment:
This should just be:

```rust
odd(name, value, params).map(|b| !b)
```

or, for clarity:

```rust
let is_odd = try!(odd(name, value, params);
!is_odd
```

I like the one-liner myself.